src/fragments/lib/graphqlapi/js/optimistic-ui.mdx

```diff
@@ -0,0 +1,195 @@
+The GraphQL API category can be used with [TanStack Query](https://tanstack.com/query/latest/docs/react/overview) to implement optimistic UI, allowing CRUD operations to be rendered immediately on the UI before the request roundtrip has completed.
+
+To get started, run the following in an existing Amplify project:
+
+```bash
+# Install TanStack Query
+npm i @tanstack/react-query
+
+# Select default configuration
+amplify add api              
+```
+
+When prompted, use the following schema, which can also be found under `amplify/backend/api/[name of project]/schema.graphql`:
+
+```graphql
+type RealEstateProperty @model @auth(rules: [{ allow: public }]) {
+  id: ID!
+  name: String!
+  address: String
+}
+```
+
+Save the schema and run `amplify push` to deploy the changes.
+
+Next, (at the root of your project?), add the required TanStack Query imports, and create a client:
+
+```ts
+import {
+  useQuery,
+  useMutation,
+  useQueryClient,
+  QueryClient,
+  QueryClientProvider,
+} from '@tanstack/react-query'
+
+// Create a client
+const queryClient = new QueryClient()
+```
+
+Next, provide the client to your app:
+
+```ts
+<QueryClientProvider client={queryClient}>
+  <App />
+</QueryClientProvider>
+```
+
+Next, within your app, access the TanStack client:
+
+```ts 
+// Access the client
+const queryClient = useQueryClient()
+```
+
+<Callout>
+
+For more details on TanStack Query, including requirements, supported browsers, and advanced usage, see the [TanStack Query documentation](https://tanstack.com/query/latest/docs/react/overview). 
+For more on Amplify GraphQL API operations, see the [API documentation](https://docs.amplify.aws/lib/graphqlapi/authz/q/platform/js/).
```

Comment:
is this the correct link? shouldn't it be this https://docs.amplify.aws/lib/graphqlapi/mutate-data/q/platform/js/ ? Also, I think we should use a relative link here instead of absolute.